js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -2073,42 +2075,50 @@ const EditingController = modules.ViewController.inherit((function() {
             const oldEditRowIndex = that._getVisibleEditRowIndex();
 
             if(!isRowEditMode(that)) {
-                result = Deferred();
-                this.executeOperation(result, () => {
-                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData);
-                    result.resolve();
+                const deferred = new Deferred();
+                result = new Deferred();
+                this.executeOperation(deferred, () => {
+                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData).always(result.resolve);
                 });
             }
+
             return result.promise();
         },
 
         _closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData) {
-            const that = this;
-            const editMode = getEditMode(that);
-            const dataController = that._dataController;
+            const editMode = getEditMode(this);
+            const dataController = this._dataController;
+            const deferred = new Deferred();
+            const promise = deferred.promise();
 
-            if(editMode === EDIT_MODE_CELL && that.hasChanges()) {
+            if(editMode === EDIT_MODE_CELL && this.hasChanges()) {
                 if(!withoutSaveEditData) {
-                    that.saveEditData().done(function(error) {
-                        if(!that.hasChanges()) {
-                            that.closeEditCell(!!error);
+                    this.saveEditData().done((error) => {
+                        if(!this.hasChanges()) {
+                            this.closeEditCell(!!error).always(deferred.resolve);
```

Comment:
`closeEditCell` can not be rejected, why are you using `always` instead of `done`?

Reply:
The cause is the same as in the previous comment)